mmdet/evaluation/metrics/crowdhuman_metric.py

```diff
@@ -0,0 +1,914 @@
+        sort of dtbox.score
+        """
+        if self.dtboxes is None:
+            return list()
+        dtboxes = self.dtboxes
+        gtboxes = self.gtboxes if self.gtboxes is not None else list()
+        dtboxes.sort(key=lambda x: x.score, reverse=True)
+        gtboxes.sort(key=lambda x: x.ign)
+
+        scorelist = list()
+        for i, dt in enumerate(dtboxes):
+            maxpos = -1
+            maxiou = thres
+
+            for j, gt in enumerate(gtboxes):
+                overlap = dt.iou(gt)
+                if overlap > maxiou:
+                    maxiou = overlap
+                    maxpos = j
+
+            if maxpos >= 0:
+                if gtboxes[maxpos].ign == 0:
+                    gtboxes[maxpos].matched = 1
+                    dtboxes[i].matched = 1
+                    scorelist.append((dt, self.ID))
+                else:
+                    dtboxes[i].matched = -1
+            else:
+                dtboxes[i].matched = 0
+                scorelist.append((dt, self.ID))
+        return scorelist
+
+
+class Database(object):
+    """Loading information and evaluation.
+
+    Args:
+        gt_path (str): Path of an annotations file.
+        dt_path (str): Path of detection results.
+        body_key (str, None): key of detection body box.
+            Valid when loading detection results and mode!=1.
+        head_key (str, None): key of detection head box.
+            Valid when loading detection results and mode!=0.
+        mode (int): Select the mode of evaluate. Valid mode include
+            0(just body box), 1(just head box) and 2(both of them).
+            Defaults to 0.
+    Examples:
+        >>> database = Database(
+        >>>     '/data/YuYoujiang/dataset/CrowdHuman/annotation_val.odgt',
+        >>>     '/data/YuYoujiang/CrowdDet/mmdetection/tools/result_meg.json',
+        >>>     'box', None, 0)
+        >>> database.compare()
+        >>> eval_results = OrderedDict()
+        >>> AP, recall, _ = database.eval_AP()
+        >>> eval_results['mAP'] = float(f'{round(AP, 3)}')
+        >>> eval_results['recall'] = float(f'{round(recall, 3)}')
+        >>> mMR, _ = database.eval_MR()
+        >>> eval_results['mMR'] = float(f'{round(mMR, 3)}')
+        >>> print(eval_results)
+    """
+
+    def __init__(self,
+                 gt_path=None,
+                 dt_path=None,
+                 body_key=None,
+                 head_key=None,
+                 mode=0):
+
+        self.images = dict()
+        self.eval_mode = mode
+        self.loadData(gt_path, body_key, head_key, if_gt=True)
+        self.loadData(dt_path, body_key, head_key, if_gt=False)
+
+        self._ignNum = sum([self.images[i]._ignNum for i in self.images])
+        self._gtNum = sum([self.images[i]._gtNum for i in self.images])
+        self._imageNum = len(self.images)
+        self.scorelist = None
+
+    def loadData(self, fpath, body_key=None, head_key=None, if_gt=True):
```

Comment:
Use lowercase for function names. e.g. `load_data`.

And I prefer to rename this method to `load_gt` or `load_ann`.